llama.cpp

```diff
@@ -910,6 +869,24 @@ struct llama_mmap {
         }
     }
 
+    void unmap(size_t offset, size_t len) {
+        int page_size = sysconf(_SC_PAGESIZE);
+        align_offset(&offset, &len, page_size);
+        if (len < (size_t)page_size) {
+            return;
+        }
+
+        void * next_page_start = (uint8_t *) addr + offset;
+        // unmap and discard the pages
+        if (munmap(next_page_start, len)) {
+            fprintf(stderr, "warning: munmap failed: %s\n", strerror(errno));
+        }
+        if (posix_madvise(next_page_start, len, POSIX_MADV_DONTNEED)) {
+            fprintf(stderr, "warning: posix_madvise(.., POSIX_MADV_DONTNEED) failed: %s\n",
+                    strerror(errno));
+        }
```

Comment:
This error triggers on macOS with `main`

```
next_page_start: 0x280000000, len: 737280
warning: posix_madvise(.., POSIX_MADV_DONTNEED) failed: Invalid argument
```

Reply:
Removed this `madvise`, it shouldn't do anything since the same range is already unmapped.